src/RadixTree.ts

```diff
@@ -1,21 +1,58 @@
 import BaseNode from './BaseNode';
 
 export default class RadixTree {
-    private roots: BaseNode[];
+    readonly root: BaseNode;
 
     constructor() {
-        this.roots = [];
+        this.root = new BaseNode('');
     }
 
     public add(str: string, data: any) {
         console.log('add', str, data);
     }
 
-    find() {
+    public find(str: string) {
+        const queue = [{ substr: str, children: this.root.getChildren(), index: -1 }];
 
-    }
+        function next() {
+            if (queue.length === 0) {
+                return null;
+            }
+            let scope = queue[queue.length - 1];
+            while (scope && (!scope.children || scope.index + 1 === scope.children.length)) {
+                queue.splice(queue.length - 1, 1);
```

Comment:
Not sure how it works internally, but potentially the `splice` may lead to several array reallocations. Though considering the potential size of the `queue` array it shouldn't be a big performance issue.

Reply:
I think that `shift()` method may be a better option here